UnitTests/CommonTestUtils/ConfigureJoinableTaskFactoryAttribute.cs

```diff
@@ -14,13 +16,25 @@ namespace CommonTestUtils
     [AttributeUsage(AttributeTargets.Assembly)]
     public sealed class ConfigureJoinableTaskFactoryAttribute : Attribute, ITestAction
     {
+        private readonly bool _checkForUpdatesOriginalValue = AppSettings.CheckForUpdates;
+
         private DenyExecutionSynchronizationContext _denyExecutionSynchronizationContext;
         private ExceptionDispatchInfo _threadException;
 
         public ActionTargets Targets => ActionTargets.Test;
 
+        static ConfigureJoinableTaskFactoryAttribute()
+        {
+            // route unhandled application exceptions to HandleApplicationThreadException
+            Application.SetUnhandledExceptionMode(UnhandledExceptionMode.CatchException);
```

Comment:
❔ This is the default value. If this line is necessary, do you know where it's being overridden with some other value?

Reply:
Since I have set `SetUnhandledExceptionMode`, I have not seen any tests hanging due to a popup "Unhandled **Application** Exception" anymore.
I don't know where it is set. Though I think this is not the default value, especially not for the test runner.

Reply:
Do you know the name of the test runner process? I can figure out from there where it's being set.

Reply:
It's `nunit3-testrunner` or the like.

Reply:
It's set here:
https://github.com/nunit/nunit-console/blob/dfd07ef27cac9082c9728f0e55b84d1237830605/src/NUnitConsole/nunit3-console/app.config#L16-L17

Reply:
Obviously, it does not work as you expect:

![grafik](https://user-images.githubusercontent.com/36601201/72480806-ddc03600-37f8-11ea-8096-46ccb21d02b8.png)

Perhaps the test assemblies would need their own `app.config`.
I'll restore it.

Reply:
Removed `Application.SetUnhandledExceptionMode` because it does not help avoiding "unhandled *application* exception" popups.
So what can we do to catch them?